### PR TITLE
Support exporting GitLab CI pipelines

### PR DESCRIPTION
**Pull request by ZhuGongpu**
Ref #226,

- Add support for exporting GitLab CI pipelines.
- Convert `DIR` and `FUNCTIONS_DIR` into relative paths in the exported pipeline.
- Remove `FUNCTIONS_DIR` existence checks to make it more flexible.
  - For example, the user may want to add an extra step in the generated pipeline to pull a kpt package which `--fn-path` points to but does not exist when exporting the pipeline.

### CODE REVIEW

**Comment by ZhuGongpu** [comment]
@Liujingfang1 @monopole @frankfarzan

File: internal/cmdexport/orchestrators/gitlabci.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	fnPaths := []string{"--fn-path"}
+
+	for _, p := range config.FnPaths {
+		// p is preprocessed in ExportRunner and guaranteed to be a relative path.
+		fnPaths = append(fnPaths, path.Join(mountDir, p))
+	}
+
+	parts := []string{
+		"docker run",
+		fmt.Sprintf("-v $PWD:%s", mountDir),
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/gitlabci.go:74]: Is `$PWD` for the root directory of a repo?

Comment (ZhuGongpu) [reply internal/cmdexport/orchestrators/gitlabci.go:74]: Yes. The pipeline runs at the root level.

File: internal/cmdexport/types/pipelineconfig.go [hunk 2]
```diff
@@ -14,8 +14,31 @@
 
 package types
 
+import "github.com/GoogleContainerTools/kpt/internal/util/pathutil"
+
 // PipelineConfig describes configuration of a pipeline.
 type PipelineConfig struct {
```
[the remark this answers is not in the document]
Comment (ZhuGongpu) [reply internal/cmdexport/types/pipelineconfig.go:20]: Good idea! In this way, we can even allow the user to specify `cwd` in the future.

File: internal/util/pathutil/pathutil.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Google LLC
```
Comment (Liujingfang1) [inline internal/util/pathutil/pathutil.go:1]: Could we move this package to "internal/cmdexport/util/path"?

File: internal/util/pathutil/pathutil.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+package pathutil
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Rel, like path.Rel, converts a path into a relative path to directory with the knowledge of cwd.
```
Comment (Liujingfang1) [inline internal/util/pathutil/pathutil.go:25]: Please add some some comments for the input of `Rel`. Are they relative path, absolute path or both?

**Pull request merged**